GitUpKit/Interface/GIGraphView.m

```diff
@@ -860,112 +860,162 @@ static void _DrawLine(GILine* line, CGContextRef context, CGFloat offset, CGFloa
+  NSMutableAttributedString* multilineAttributedTitle = [[NSMutableAttributedString alloc] initWithString:multilineTitle attributes:multilineTitleAttributes];
+  CFRelease(titleFont);
   
-  CFAttributedStringRef string = CFAttributedStringCreate(kCFAllocatorDefault, (CFStringRef)label, (CFDictionaryRef)attributes);
-  CTFramesetterRef framesetter = CTFramesetterCreateWithAttributedString(string);
-  CGSize size = CTFramesetterSuggestFrameSizeWithConstraints(framesetter, CFRangeMake(0, CFAttributedStringGetLength(string)), NULL, CGSizeMake(CGFLOAT_MAX, CGFLOAT_MAX), NULL);
+  // Change font to bold on ranges collected before
+  CTFontRef boldFont = CTFontCreateUIFontForLanguage(kCTFontUIFontEmphasizedSystem, 12.0, CFSTR("en-US"));
+  for (NSValue* bold in boldRanges) {
+    [multilineAttributedTitle addAttribute:NSFontAttributeName value:(id)boldFont range:bold.rangeValue];
+  }
+  CFRelease(boldFont);
+  
+  // Change color to dark on ranges collected before
+  NSColor* darkColor = [color shadowWithLevel:0.8];
+  for (NSValue* dark in darkRanges) {
+    [multilineAttributedTitle addAttribute:NSForegroundColorAttributeName value:darkColor range:dark.rangeValue];
+  }
+  
+  // Prepare CoreText string from the rich attributed title
+  CTFramesetterRef framesetter = CTFramesetterCreateWithAttributedString((CFAttributedStringRef)multilineAttributedTitle);
+  CGSize size = CTFramesetterSuggestFrameSizeWithConstraints(framesetter, CFRangeMake(0, multilineAttributedTitle.length), NULL, CGSizeMake(CGFLOAT_MAX, CGFLOAT_MAX), NULL);
   CGRect textRect = CGRectMake(kTitleOffsetX, kTitleOffsetY, ceil(size.width), ceil(size.height));
   CGPathRef path = CGPathCreateWithRect(textRect, NULL);
-  CTFrameRef frame = CTFramesetterCreateFrame(framesetter, CFRangeMake(0, CFAttributedStringGetLength(string)), path, NULL);
-  CFAttributedStringRef character = CFAttributedStringCreate(kCFAllocatorDefault, CFSTR("\u2026"), (CFDictionaryRef)attributes);
-  CTLineRef token = CTLineCreateWithAttributedString(character);
+  CTFrameRef frame = CTFramesetterCreateFrame(framesetter, CFRangeMake(0, multilineAttributedTitle.length), path, NULL);
+  CFAttributedStringRef ellipsis = CFAttributedStringCreate(kCFAllocatorDefault, CFSTR("\u2026"), (CFDictionaryRef)multilineTitleAttributes);
+  CTLineRef ellipsisToken = CTLineCreateWithAttributedString(ellipsis);
   
   // Prepare context
-  
   CGContextSaveGState(context);
   CGContextTranslateCTM(context, x, y);
-  CGContextRotateCTM(context, 40.0 / 180.0 * M_PI);
-  
-  // Draw label
-  
-#if __DEBUG_BOXES__
-  CGRect labelRect = CGRectInset(CGRectMake(textRect.origin.x, textRect.origin.y, MIN(textRect.size.width, kMaxBranchTitleWidth), textRect.size.height), -4.5, -3.5);
-  CGContextSetRGBFillColor(context, 1.0, 0.0, 0.0, 0.666);
-  CGContextFillRect(context, labelRect);
-#endif
+  CGContextRotateCTM(context, 45.0 / 180.0 * M_PI);
+  CGContextSetLineWidth(context, 0.5);
```

Comment:
This should move next to the call to `CGContextSetStrokeColorWithColor()`. It's kind of weird to set the width of the stroke here but its color way later.